keychain-sdk/key_requests.go

```diff
@@ -27,13 +27,17 @@ type keyResponseWriter struct {
 	onComplete   func()
 }
 
-func (w *keyResponseWriter) Fulfil(publicKey []byte) error {
+func (w *keyResponseWriter) Fulfil(publicKey []byte) (err error) {
```

Comment:
I'd use just and then return the result of the `txWriter`
```suggestion
func (w *keyResponseWriter) Fulfil(publicKey []byte) (error) {
```

Reply:
I can't do that because I read `err` inside the deferred function